### PR TITLE
*: move `session.NewStore` to `store.New` to achieve semantic accuracy

### PR DESCRIPTION
**Pull request by lonnng**
The API `session.NewStore` should be placed in package `store` and
rename `NewStore` to `New` for achieving more semantic accuracy.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8617)

### CODE REVIEW

**Comment by lonnng** [comment]
Can not access github.com/pingcap/tidb-test

```
# github.com/pingcap/tidb-test/ddl_test/ddltest [github.com/pingcap/tidb-test/ddl_test/ddltest.test]
ddltest/ddl_test.go:99:17: undefined: session.NewStore
ddltest/ddl_test.go:1029:2: undefined: session.RegisterStore
FAIL	github.com/pingcap/tidb-test/ddl_test/ddltest [build failed]
```

**Comment by winkyao** [comment]
@zimulala Please help @lonnng to fix the ddltest in tidb-test.

File: store/store_test.go [hunk 1]
```diff
@@ -637,3 +645,15 @@ func (s *testKVSuite) TestIsolationMultiInc(c *C) {
...
+func (s *testKVSuite) TestRetryOpenStore(c *C) {
+	begin := time.Now()
+	Register("dummy", &brokenStore{})
+	store, err := newStoreWithRetry("dummy://dummy-store", 3)
+	if store != nil {
+		defer store.Close()
+	}
+	c.Assert(err, NotNil)
+	elapse := time.Since(begin)
+	c.Assert(uint64(elapse), GreaterEqual, uint64(3*time.Second))
```
Comment (winkyao) [inline store/store_test.go]: Can we reduce the test executing time?

Comment (winkyao) [reply store/store_test.go]: Fine, you just move the code..

**Comment by ngaut** [comment]
LGTM

**Pull request merged**